src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -110,5 +110,45 @@ public List<CheepDTO> GetCheepsFromAuthor(string author, int? pageNum)
         }
     }
 
+    public int GetCountOfAllCheeps()
+    {
+        //Creates a list of max 32 CheepDTO sorted by recent cheep
+
+        List<CheepDTO> cheepsToReturn = new List<CheepDTO>();
+
+        var cheepsDTO = db.Cheeps.OrderByDescending(c => c.TimeStamp.Ticks).Select(CheepDTO => new CheepDTO
+        {
+            //Sets the properties of the Cheep
+            AuthorId = CheepDTO.Author.AuthorId,
+            Author = CheepDTO.Author.Name,
+            Message = CheepDTO.Text,
+            Timestamp = CheepDTO.TimeStamp
+        }
+        );
+
+        cheepsToReturn.AddRange(cheepsDTO);
+        return cheepsToReturn.Count();
+    }
+
+    public int GetCountOfAllCheepFromAuthor(string author)
+    {
+        //Creates a list of max 32 CheepDTO sorted by recent cheep and only for the given author
+
+        List<CheepDTO> cheepsToReturn = new List<CheepDTO>();
+
+        var cheepsDTO = db.Cheeps.OrderByDescending(c => c.TimeStamp.Ticks)
```

Comment:
Could we perphaps request the count, via an aggregate, function directly instead.
If we imageine we have thousands of cheeps, a lot of data is sent between the layers, were by returning the count only, a constant low amount of data is always sent.

Reply:
The count now comes from the SQL query as requested